lib/trino-parquet/src/main/java/io/trino/parquet/writer/PrimitiveColumnWriter.java

```diff
@@ -257,21 +256,21 @@ private List<ParquetDataOutput> getDataStreams()
         // write dict page if possible
         DictionaryPage dictionaryPage = primitiveValueWriter.toDictPageAndClose();
         if (dictionaryPage != null) {
-            long uncompressedSize = dictionaryPage.getUncompressedSize();
+            int uncompressedSize = dictionaryPage.getUncompressedSize();
             byte[] pageBytes = dictionaryPage.getBytes().toByteArray();
             ParquetDataOutput pageData = compressor != null
                     ? compressor.compress(pageBytes)
                     : createDataOutput(Slices.wrappedBuffer(pageBytes));
-            long compressedSize = pageData.size();
+            int compressedSize = pageData.size();
 
             ByteArrayOutputStream dictStream = new ByteArrayOutputStream();
             parquetMetadataConverter.writeDictionaryPageHeader(
-                    toIntExact(uncompressedSize),
```

Comment:
> Allows simplifying some of the code working with pages sizes
to use ints instead of longs

I am not sure this is the right direction. Why `int` is better for data size?

At least i think https://github.com/trinodb/trino/pull/18479 is going in the opposite direction. (now added you as a reviewer there)

Reply:
All the underlying values here are ints, it's the usage of `long BytesInput#size` from parquet lib that was forcing us to use longs and then cast them back to int earlier. Now that has been cleaned up and it's no longer necessary to use a long. Everything implementing `ParquetDataOutput#size` is now actually returning an `int`, so storing it in a long and then casting to int is unnecessary.

Reply:
are we limited to 2 GB?

Reply:
yup, we have a check in ParquetWriterConfig to limit `parquet.writer.block-size` to 2GB. The standard itself allows larger row groups, but it is not something we want users to do.